test/runtests.jl

```diff
@@ -34,24 +34,40 @@ istest(f) = endswith(f, ".jl") && startswith(f, "test_")
 testfiles = sort(filter(istest, readdir(testdir)))
 
 @testset "$f" for f in testfiles
-    mpiexec() do cmd
+    mpiexec() do mpirun
+        cmd = `$mpirun -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`
         if f == "test_spawn.jl"
-            run(`$cmd -n 1 $(Base.julia_cmd()) $(joinpath(testdir, f))`)
+            # Some command as the others, but always use a single process
+            _cmd = deepcopy(cmd)
+            _cmd.exec[findfirst(==("$nprocs"), _cmd.exec)] = "1"
+            run(_cmd)
         elseif f == "test_threads.jl"
             withenv("JULIA_NUM_THREADS" => "4") do
-                run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`)
+                run(cmd)
             end
         elseif f == "test_error.jl"
-            r = run(ignorestatus(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`))
+            r = run(ignorestatus(cmd))
             @test !success(r)
+        elseif f == "test_errorhandler.jl" && (MPI.identify_implementation()[1] == MPI.UnknownMPI ||
+            # Fujitsu MPI is known to not work with custom error handlers
+            startswith(MPI.MPI_LIBRARY_VERSION_STRING, "FUJITSU MPI"))
```

Comment:
Would this be OK for the time being?  We allow failures for unknown MPIs and Fujitsu MPI, but the error message is still shown.

Reply:
Yeah I am okay with this!